frontend/src/metabase-lib/lib/Question.ts

```diff
@@ -1021,11 +1021,16 @@ class QuestionInner {
   }
 
   dependentMetadata(): DependentMetadataItem[] {
-    if (!this.isDataset()) {
```

Comment:
Seems like we can open up `Question.prototype.dependentMetadata` to all Questions now, and not just datasets. There's an e2e test associated with the bug that this fixed and it still passes so 🤷 

We need to ensure that the underlying table associated with a Question is fetched, even when the Question uses a nested query so that we always have access to the real, concrete fields. Also, because structured query's secretly check access to the "root" table in order to determine if a user has permissions to save questions, interestingly.